core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -148,13 +148,26 @@ private[spark] class MesosSchedulerBackend(
         Value.Scalar.newBuilder()
           .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
       .build()
-    MesosExecutorInfo.newBuilder()
+    val executorInfo = MesosExecutorInfo.newBuilder()
       .setExecutorId(ExecutorID.newBuilder().setValue(execId).build())
       .setCommand(command)
       .setData(ByteString.copyFrom(createExecArg()))
       .addResources(cpus)
       .addResources(memory)
-      .build()
+
+    sc.conf.getOption("spark.mesos.executor.docker.image").map { image:String =>
+      val container = executorInfo.getContainerBuilder()
+      val volumes = sc.conf
+        .getOption("spark.mesos.executor.docker.volumes")
+        .map(MesosSchedulerBackendUtil.parseVolumesSpec)
+      val portmaps = sc.conf
+        .getOption("spark.mesos.executor.docker.portmaps")
+        .map(MesosSchedulerBackendUtil.parsePortMappingsSpec)
+      MesosSchedulerBackendUtil.withDockerInfo(
+        container, image, volumes = volumes, portmaps = portmaps)
+    }
```

Comment:
This chunk of code is duplicated. You should factor this into a method in your utils class, and maybe call it `setupDockerContainer(conf: SparkConf)`